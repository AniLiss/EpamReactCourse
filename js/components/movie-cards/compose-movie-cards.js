import React from 'react';
import {MovieCard} from '../movie-card/movie-card';

export const composeMovieCards = (data) => {
    console.log('data', data);
    return data.map((item, index) =>
         <MovieCard
            key={index}
            title={item.title}
            releaseDate={item.release_date}
            genre={item.genre}
        />
    );
};
