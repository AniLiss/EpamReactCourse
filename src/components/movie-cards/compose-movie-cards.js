import React from 'react';
import {MovieCard} from '../movie-card/movie-card';

export const composeMovieCards = (data) => {
    return data.map((item, index) =>
         <MovieCard
            key={index}
            title={item.title}
            releaseDate={item.release_date}
            genre={item.genre}
        />
    );
};
