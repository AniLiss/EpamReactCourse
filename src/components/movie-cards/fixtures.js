export const mockData = [
    {
        title: 'Kill Bill. Vol. 1',
        release_date: 2003,
        genre: 'Action & Adventure'
    },
    {
        title: 'Kill Bill. Vol. 2',
        release_date: 2004,
        genre: 'Action & Adventure'
    },
    {
        title: 'Pulp Fiction',
        release_date: 1994,
        genre: 'Oscar-winning Movie'
    }
];
