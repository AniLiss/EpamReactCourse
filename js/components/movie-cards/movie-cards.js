import React from 'react';
import {composeMovieCards} from './compose-movie-cards';

const mockData = [
    {
        title: 'Kill Bill. Vol. 1',
        release_date: 2003,
        genre: 'Action & Adventure'
    },
    {
        title: 'Kill Bill. Vol. 2',
        release_date: 2004,
        genre: 'Action & Adventure'
    },
    {
        title: 'Pulp Fiction',
        release_date: 1994,
        genre: 'Oscar-winning Movie'
    }
];

export const MovieCards = () => {
    return (<div className="movies-cards">
        {composeMovieCards(mockData)}
    </div>)
};
