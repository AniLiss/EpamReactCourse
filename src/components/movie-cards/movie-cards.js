import React from 'react';
import {composeMovieCards} from './compose-movie-cards';
import {mockData} from './fixtures';

export const MovieCards = () => {
    return (<div className="movies-cards">
        {composeMovieCards(mockData)}
    </div>)
};
