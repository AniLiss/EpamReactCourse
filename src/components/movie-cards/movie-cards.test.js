import React from 'react';
import {shallow} from 'enzyme';
import { MovieCards } from './movie-cards';

const wrapper = shallow(<MovieCards />);

describe('MovieCards', () => {
    it('is expected to render MovieCards correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
});
