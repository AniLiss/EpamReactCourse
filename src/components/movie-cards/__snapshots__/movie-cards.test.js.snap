// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCards is expected to render MovieCards correctly 1`] = `
<div
  className="movies-cards"
>
  <MovieCard
    genre="Action & Adventure"
    key="0"
    releaseDate={2003}
    title="Kill Bill. Vol. 1"
  />
  <MovieCard
    genre="Action & Adventure"
    key="1"
    releaseDate={2004}
    title="Kill Bill. Vol. 2"
  />
  <MovieCard
    genre="Oscar-winning Movie"
    key="2"
    releaseDate={1994}
    title="Pulp Fiction"
  />
</div>
`;
